dnsapi/dns_hosttech.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/bash
+
+#Hosttech_Key="asdfasdfawefasdfawefasdafe"
+
+Hosttech_Api="https://api.ns1.hosttech.eu/api/user/v1"
+
+########  Public functions #####################
+
+#Usage: add  _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_hosttech_add() {
+  fulldomain=$1
+  txtvalue=$2
+
+  Hosttech_Key="${Hosttech_Key:-$(_readaccountconf_mutable Hosttech_Key)}"
+  if [ -z "$Hosttech_Key" ]; then
+    Hosttech_Key=""
+    _err "You didn't specify a Hosttech api key"
+    _err "You can get yours from https://www.myhosttech.eu/user/dns/api"
+    return 1
+  fi
+
+  #save the api key and email to the account conf file.
+  _saveaccountconf_mutable Hosttech_Key "$Hosttech_Key"
+
+  _debug "First detect the root zone"
+  if ! _get_root "$fulldomain"; then
+    _err "invalid domain"
+    return 1
+  fi
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  _info "Adding record"
+  if _hosttech_rest POST "zones/$_domain/records" "{\"type\":\"TXT\",\"name\":\"$_sub_domain\",\"text\":\"$txtvalue\",\"ttl\":600}"; then
+    if _contains "$_response" "$_sub_domain"; then
+      _debug recordID "$(echo "$_response" | grep -o '"id":[^"]*' | grep -Po "\d+")"
+
+      #save the created recordID to the account conf file, so we can read it back for deleting in dns_hosttech_rm.
+      _saveaccountconf recordID "$(echo "$_response" | grep -o '"id":[^"]*' | grep -Po "\d+")"
```

Comment:
don't save it.
you should be quering the recordId when you are about to delete the records.